app/org/sagebionetworks/bridge/hibernate/HibernateAccountDao.java

```diff
@@ -403,27 +394,38 @@ private String hashCredential(PasswordAlgorithm algorithm, String type, String v
         return hash;
     }
 
-    // Helper method to get a single account for a given study and email address.
-    private HibernateAccount getHibernateAccountByEmail(StudyIdentifier studyId, String email) {
-        List<HibernateAccount> accountList = hibernateHelper.queryGet("from HibernateAccount where studyId='" +
-                studyId.getIdentifier() + "' and email='" + email + "'", null, null, HibernateAccount.class);
+    // Helper method to get a single account for a given study and id, email address, or phone number.
+    private HibernateAccount getHibernateAccount(AccountId accountId) {
+        // This is the only method where accessing null values is not an error, since we're searching for 
+        // the value that was provided. There will be one.
+        AccountId unguarded = accountId.getUnguardedAccountId();
+        if (unguarded.getId() != null) {
+            return hibernateHelper.getById(HibernateAccount.class, unguarded.getId());
+        }
+        String query = null;
+        if (unguarded.getEmail() != null) {
+            query = String.format(EMAIL_QUERY, unguarded.getStudyId(), unguarded.getEmail());
+        } else {
+            query = String.format(PHONE_QUERY, unguarded.getStudyId(), unguarded.getPhone().getNumber());
+        }
+        List<HibernateAccount> accountList = hibernateHelper.queryGet(query, null, null, HibernateAccount.class);
         if (accountList.isEmpty()) {
             return null;
         }
         HibernateAccount hibernateAccount = accountList.get(0);
-
         if (accountList.size() > 1) {
-            LOG.warn("Multiple accounts found for the same study and email address; example accountId=" +
-                    hibernateAccount.getId());
+            LOG.warn("Multiple accounts found email/phone query; example accountId=" + hibernateAccount.getId());
         }
-
         return hibernateAccount;
     }
 
     /** {@inheritDoc} */
     @Override
-    public void deleteAccount(Study study, String id) {
-        hibernateHelper.deleteById(HibernateAccount.class, id);
+    public void deleteAccount(AccountId accountId) {
+        HibernateAccount hibernateAccount = getHibernateAccount(accountId);
+        if (hibernateAccount != null) {
+            hibernateHelper.deleteById(HibernateAccount.class, hibernateAccount.getId());    
+        }
```

Comment:
This is functionally identical, but in the new code, in the string ID case, we'll query the DB for an account, and then only use the string ID (which we already have). Is it worth adding a hasId() to AccountId and optimizing on the case where we only need the string ID?